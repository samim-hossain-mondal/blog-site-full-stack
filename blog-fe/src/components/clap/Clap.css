button{
    padding: 3px;
    border: none;
    background-color: transparent;
}

.clap{
    display: flex;
    flex-direction: row;
}

.clap>h{
    font-size: 15px;
    padding: 5px;
    padding-top: 7px;
}