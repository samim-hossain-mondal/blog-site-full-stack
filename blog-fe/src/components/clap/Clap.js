import React, { useState } from 'react';
import clapping from '../../assets/icons/clapping.svg';
import './Clap.css';

export default function Clap(props) {
  const [clap, setClap] = useState(0);
  const clapHandler = () => {
    if (clap === 0) {
      setClap(clap + 1);
    } else {
      setClap(clap - 1);
    }
  };
  return (
    <div className='clap'>
      <img className="icons" data-testid="clap-btn" onClick={clapHandler}
        style={{ ...(clap === 0 && { opacity: 0.35 }) }} src={clapping} />
      <h> {props.claps + clap} </h>
    </div>
  );
}