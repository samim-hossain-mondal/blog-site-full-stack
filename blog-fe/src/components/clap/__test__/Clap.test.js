/* eslint-disable no-undef */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Clap from '../Clap';

describe('Clap', () => {
  it('should render', () => {
    const { asFragment } = render(<Clap />);
    expect(asFragment()).toMatchSnapshot();
  });
  describe('when the user clicks on the clap button', () => {
    it('should increment the clap count', () => {
      const { queryByTestId } = render(<Clap claps={0} />);
      const clapBtn = queryByTestId('clap-btn');
      fireEvent.click(clapBtn);
      const claps = queryByTestId('clap-btn');
      expect(claps).toBeTruthy();
    });
    it('should decrement the clap count', () => {
      const { queryByTestId } = render(<Clap claps={1} />);
      const clapBtn = queryByTestId('clap-btn');
      fireEvent.click(clapBtn);
      const claps = queryByTestId('clap-btn');
      expect(claps).toBeTruthy();
    });
  });
});