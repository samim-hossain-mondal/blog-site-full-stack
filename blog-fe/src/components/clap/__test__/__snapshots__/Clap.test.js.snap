// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clap should render 1`] = `
<DocumentFragment>
  <div
    class="clap"
  >
    <img
      class="icons"
      data-testid="clap-btn"
      src="clapping.svg"
      style="opacity: 0.35;"
    />
    <h>
       NaN 
    </h>
  </div>
</DocumentFragment>
`;
